### PR TITLE
add design-system v2.0.2 with npm auto-update

### PR DESCRIPTION
**Pull request by maruilian11**
**git repo url:** https://github.com/salesforce-ux/design-system
**Watch 125
  Star 1,122
 Fork 178**
@Amomo could u help me check this PR for #6573 ? thank you!

**here are my steps:
regard** `/assets` **as basepath
remove *.txt and *.html**
**use**  `find . -type f | perl -ne 'print $1 if m/\.([^.\/]+)$/' | sort -u`  **to check all file types
set up auto-update filemap**

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.
# Essential checklist
- [ ] I'm the author of this library
  - [ ] I would like to add link to the page of this library on CDNJS on website / readme
- [x] This lib was not found on cdnjs repo
- [x] No already exist / duplicated issue and PR 
- [x] The lib has notable popularity 
  - [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
  - [ ] More than 500 downloads stats per month on npm registry 
- [x] Project has public repository on famous online hosting platform (or been hosted on npm)
# Auto-update checklist
- [x] Has valid tags for each versions (for git auto-update) 
- [x] Auto-update setup
- [x] Auto-update target/source is valid.
- [x] Auto-update filemap is correct.
# Git commit checklist
- [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
- [x] The parent of the commit(s) in the PR is not old than 3 days.
- [x] Pull request is sending from a non-master branch with meaningful name.
- [x] Separate unrelated changes into different commits.
- [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
- [x] Close corresponding issue in commit message
- [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by PeterDaveHello** [comment]
There are 1886 files under `assets`, I'm wondering if we can just use the whole `assets`, any info about the difference between of using `**/*` under assets and using current filemap?

**Comment by maruilian11** [comment]
yes, there are 
`fonts/License-for-font.txt` 
`icons/License-for-icons.txt` 
`icons/README`
`images/License-for-images.txt`
`icons/**/symbols.html`.

**Comment by PeterDaveHello** [comment]
Maybe we could just exclude them.

**Comment by maruilian11** [comment]
use `!(**/+(*.+(txt|html)|README))` to exclude.

**Comment by maruilian11** [comment]
ping @Amomo

**Comment by Amomo** [comment]
@maruilian11 sorry for late, `v2.0.3` has come, could you update this PR?
Thank you very much! 😭

**Comment by PeterDaveHello** [comment]
@Piicksarn please also help review this PR, thanks.

**Comment by Piicksarn** [comment]
@maruilian11, if u update the pr, i would help u for checkout :wink:

**Comment by maruilian11** [comment]
@PeterDaveHello I found some auto-update execution problem, for example:
filemap work with `**/*.+(txt|html)`, but it can't work with `!(**/*.+(txt|html))`,
which means `!(**/+(*.+(txt|html)|README))` can't work, either.

**Comment by PeterDaveHello** [comment]
So is current config working?

**Comment by maruilian11** [comment]
@PeterDaveHello I just found out a big problem.
`**/!(*+(html|txt|md))` will only exclude the match pattern in basepath, which is also the way we mention in [autoupdate.md](https://github.com/cdnjs/cdnjs/blob/master/documents/autoupdate.md).
Maybe we should remove the exclude example temporarily in autoupdate.md because it doesn't work in this case. Sorry for I said I test the rules. I can't remember which lib I test the rule.

I try many ways to exclude one certain type under all directories by use `!(...)`
`**/*.+(html|txt|md)` can catch all html, txt and md type files.
But `!(**/*.+(html|txt|md)` CAN'T catch all files except all html, txt and md type files.
I think that Because there must be **pattern** in `!(...)`, and `**/*.html` is not a pattern. (only `*.html` is)
I try `**/*/!(*.+(html|txt|md))` which will only exclude `*/*.+(html|txt|md)` but there are still `*/*/*.html` there.
and I try `**/*/*/!(*.+(html|txt|md))` which will remove `*/*/*.html`, but directories which has only one subdirectory will be exclude.
the best rule I success catch the files we need is `**/+(*.css|*.eot|*.gif|*.jpg|*.png|*.svg|*.ttf|*.woff|*.woff2)`, but I think it's not a good solution.

I search google for help.
there are some similar questions people asked:
http://stackoverflow.com/questions/23809897/node-js-glob-pattern-for-excluding-multiple-files
https://github.com/gulpjs/gulp/issues/165 

But seem that the only way to let `!**/*.+(html|txt|md)` rule work is improve our [autoupdate.js](https://github.com/cdnjs/cdnjs/blob/master/auto-update.js)

I will keep research this issue.

**Comment by PeterDaveHello** [comment]
Any updates here? Thanks!

**Comment by maruilian11** [comment]
@PeterDaveHello Can I set assets folder of npm package as basepath and use `**/*` as filemap in this case?

**Comment by PeterDaveHello** [comment]
so what's the point to transfer to use `**/*` here?

**Comment by maruilian11** [comment]
@PeterDaveHello Because there are many files under assets, and I think we need almost all files under it, except `.md`, `.txt` and `.html` which will be ignored by our `.gitignore`.

**Comment by PeterDaveHello** [comment]
Would @pvnr0082t like to give any suggestions? Thanks.

**Comment by maruilian11** [comment]
ping @pvnr0082t thanks!

**Comment by sufuf3** [comment]
`"!(**/+(*.+(txt|html)|README))"` still can't get files from newest versions. 😞 
`**/+(*.css|*.eot|*.gif|*.jpg|*.png|*.svg|*.ttf|*.woff|*.woff2)` is a good way, but if you think it's not a good solution.
What I try as the followings:
```
"+(fonts|images|styles)/**/!(*.txt|*.md|*.html)",
"icons/**/!(*.txt*|*.md*|*.html|*-sprite)"
```
it's OK for auto-update.

Or I think maybe we could also exclude `icons/*-sprite/` folder. it's just the `symbols.html` we don't need, and `svg/symbols.svg` is used by `symbols.html`.

Maybe this pattern needs to discuss.

**Comment by PeterDaveHello** [comment]
@sufuf3 pattern needs to discuss or pattern needs to be discussed?

**Comment by sufuf3** [comment]
Sorry for my poor English.
The pattern needs to be discussed.

**Comment by maruilian11** [comment]
hmmm... the issue I want to discuss is Can we use `**/*` in this case?

>@PeterDaveHello Can I set assets folder of npm package as basepath and use **/* as filemap in this case?

>so what's the point to transfer to use **/* here?

>@PeterDaveHello Because there are many files under assets, and I think we need almost all files under it, except .md, .txt and .html which will be ignored by our .gitignore.

Above pattern seems too complicated?

**Comment by maruilian11** [comment]
ping @cdnjs/library-reviewer @sufuf3 for some advice, thanks.

**Comment by sufuf3** [comment]
How about use `"+(fonts|images|styles|icons)/**/!(*.txt|*.md|*.html|*-sprite|README)"`?
this can get the files we want.

**Comment by maruilian11** [comment]
@sufuf3 Thank you. And I want to ask the reason we need to exclude `*-sprite`?

@PeterDaveHello There are 3 licenses under this lib, `BSD-3-Clause`, `CC-BY-ND-4.0` and their customized [`font license`](https://github.com/salesforce-ux/licenses/blob/master/LICENSE-font.txt). Can we use `font license` in licenses? 
https://github.com/salesforce-ux/design-system#licenses

**Comment by sufuf3** [comment]
I'd like to exclude `action-sprite`, `custom-sprite`, `doctype-sprite`, `standard-sprite`, `utility-sprite` folders because I think the file under folders we don't need to host.
but it seems that this pattern just excludes the `symbols.html` not exclude `symbols.svg`.

**Comment by sufuf3** [comment]
@PeterDaveHello Can we use font license in licenses?
As this comment said https://github.com/cdnjs/cdnjs/pull/8358#issuecomment-322703261
Thank you.

**Comment by PeterDaveHello** [comment]
@sufuf3 See it it exist in the license list.

**Comment by sufuf3** [comment]
@maruilian11 please take a look at this comment https://github.com/cdnjs/cdnjs/pull/8358#issuecomment-328336537
Thank you.

**Comment by maruilian11** [comment]
@cdnjs/library-reviewer @sufuf3 I modify the filemap. And `font license` isn't in license list, so I didn't add it.

File: ajax/libs/design-system/package.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
+  "name": "design-system",
+  "filename": "styles/salesforce-lightning-design-system.min.css",
+  "description": "Salesforce Lightning Design System",
+  "version": "2.0.2",
+  "author": "Salesforce",
+  "license": "BSD-2-Clause",
```
Comment (extend1994) [inline ajax/libs/design-system/package.json]: The licenses are BSD-3-Clause, CC-BY-ND-4.0.
Ref: https://github.com/salesforce-ux/design-system#licenses.

File: ajax/libs/design-system/package.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/salesforce-ux/design-system.git"
+  },
+  "keywords": [
+    "Salesforce",
+    "Lightning"
+  ],
+  "homepage": "https://www.lightningdesignsystem.com",
+  "npmName": "@salesforce-ux%2fdesign-system",
```
[the remark this answers is not in the document]
Comment (maruilian11) [reply ajax/libs/design-system/package.json]: Actually it can run the config, too. 😂

**Comment by maruilian11** [comment]
@cdnjs/library-reviewer @sufuf3 I updated license and npm lib name. Please help me review this PR, thanks!

File: ajax/libs/design-system/package.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
+{
+  "name": "design-system",
+  "filename": "styles/salesforce-lightning-design-system.min.css",
+  "description": "Salesforce Lightning Design System",
+  "version": "2.0.2",
+  "author": "Salesforce",
+  "license": "BSD-3-Clause",
```
Comment (extend1994) [inline ajax/libs/design-system/package.json]: Missing CC-BY-ND-4.0.
It would be
```
licenses: [
  "BSD-3-Clause",
  "CC-BY-ND-4.0"
]
```

**Comment by maruilian11** [comment]
@extend1994 Thanks! How stupid mistake I made..

**Comment by maruilian11** [comment]
Thanks.

**Comment by PeterDaveHello** [comment]
I doubt if you tagged the wrong guy in commit message @maruilian11 ?

**Comment by maruilian11** [comment]
@stefsullrew I think tag you in this PR would be better. We are now starting provide the CDN service for your popular lib. :smiley: 
https://cdnjs.com/libraries/design-system

**Comment by stefsullrew** [comment]
You're aware this is a very old version of the Design System? The last publicly released version is 2.4.x ... What is the purpose of the CDN?

**Comment by PeterDaveHello** [comment]
@stefsullrew see https://cdnjs.com/libraries/design-system/2.4.3

**Comment by DrBoolean** [comment]
@stefsullrew cdnjs rules. Can be very useful with codepens and other things to just bring in a library on the fly.

**Pull request merged**